### ISSUE TITLE
args[module] error for templated module parameter

### ISSUE THREAD

**Issue by jantari** [ansible/ansible-lint#2837]
##### Summary

The new args[module] rule reports errors if you use a Jinja template for a module parameter.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.10.0 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Create the following directory structure:

```
.
└── tasks
    └── task.yml
```

with the `task.yml` file content:

```yaml
# Dummy example data
- name: Get WAN interface(s)
  ansible.builtin.set_fact:
    _usable_wan_interfaces:
      - name: wan1
      - name: wan2

- name: Add VPN interface to RZ zone
  fortios_system_zone:
    vdom: "{{ vdom }}"
    state: "present"
    access_token: "{{ fortios_access_token | default(omit) }}"
    system_zone:
      name: RZ
      description: VPN interfaces going to HLB RZ
      interface: >
        [ {% for item in _usable_wan_interfaces %} { 'interface_name': 'INS2S-{{ item.name }}' }, {% endfor %} ]
      intrazone: deny
```

and run `ansible-lint`.

##### Desired Behavior

Ideally it would just pass, but according to #2833 and [the docs](https://ansible-lint.readthedocs.io/rules/args/) warnings may be expected due to known limitations.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

ansible-lint reports a FATAL error:

```
WARNING  Listing 1 violation(s) that are fatal
args[module]: dictionary requested, could not parse JSON or key=value
tasks/task.yml:8 Task/Handler: Add VPN interface to RZ zone
```
